doc/whats_new/v0.20.rst

```diff
@@ -26,6 +26,8 @@ random sampling procedures.
 - :class:`linear_model.OrthogonalMatchingPursuit` (bug fix)
 - :class:`metrics.roc_auc_score` (bug fix)
 - :class:`metrics.roc_curve` (bug fix)
+- :class:`mixture.BayesianGaussianMixture` (bug bix)
```

Comment:
Remind me: does this pr actually change  the prediction? If it's only affecting the attribute value, I think we should leave it out of here

Reply:
Yes, it may change the prediction in some very rare case. Consider:

```python
gm = GaussianMixture(warm_start=True, max_iter=10)
gm.fit(X1)
gm.fit(X2)
```

If `X2` is different from `X1`, is does not make sense to use the previous value of `lower_bound_` as a starting point. The previous implementation would do that, and therefore it might wrongly detect convergence after the first iteration (this is unlikely, but possible, in particular if `X2` is very similar to `X1`, or if `tol` is large). This PR fixes this. Thus, the fixed algorithm might converge to a better solution and produce different (better) predictions.

You could consider these as separate issues: (1) wrong `lower_bound_` value, and (2) wrong convergence detection logic. However, they're both caused by the same few lines of code, so I fixed them both in this one PR.

Reply:
Thanks for the explanation